modules/meshrenderinggl/src/processors/sphererasterizer.cpp

```diff
@@ -0,0 +1,264 @@
+    , forceOpaque_("forceOpaque", "Shade Opaque", false, InvalidationLevel::InvalidResources)
+    , useUniformAlpha_("useUniformAlpha", "Uniform Alpha", false,
+                       InvalidationLevel::InvalidResources)
+    , uniformAlpha_("alphaValue", "Alpha", 0.7f, 0, 1, 0.1f, InvalidationLevel::InvalidOutput)
+    , clipping_("clipping", "Clipping")
+    , clipMode_("clipMode", "Clip Mode",
+                {{"discard", "Discard Glyph", GlyphClippingMode::Discard},
+                 {"cut", "Cut Glypyh", GlyphClippingMode::Cut}},
+                0, InvalidationLevel::InvalidResources)
+    , clipShadingFactor_("clipShadingFactor", "Clip Surface Adjustment", 0.9f, 0.0f, 2.0f)
+    , shadeClippedArea_("shadeClippedArea", "Shade Clipped Area", false,
+                        InvalidationLevel::InvalidResources)
+    , sphereProperties_("sphereProperties", "Sphere Properties")
+    , forceRadius_("forceRadius", "Force Radius", false, InvalidationLevel::InvalidResources)
+    , defaultRadius_("defaultRadius", "Default Radius", 0.05f, 0.00001f, 2.0f, 0.01f)
+    , forceColor_("forceColor", "Force Color", false, InvalidationLevel::InvalidResources)
+    , defaultColor_("defaultColor", "Default Color", vec4(0.7f, 0.7f, 0.7f, 1.0f), vec4(0.0f),
+                    vec4(1.0f), vec4(0.01f), InvalidationLevel::InvalidOutput,
+                    PropertySemantics::Color)
+    , useMetaColor_("useMetaColor", "Use meta color mapping", false,
+                    InvalidationLevel::InvalidResources)
+    , metaColor_("metaColor", "Meta Color Mapping")
+
+    , camera_("camera", "Camera", util::boundingBox(inport_))
+    , trackball_(&camera_)
+    , lighting_("lighting", "Lighting", &camera_)
+    , transformSetting_("transformSettings", "Additional Transform")
+    , shaders_{new MeshShaderCache(
+          {{ShaderType::Vertex, std::string{"sphereglyph.vert"}},
+           {ShaderType::Geometry, std::string{"sphereglyph.geom"}},
+           {ShaderType::Fragment, std::string{"oit-sphereglyph.frag"}}},
+
+          {{BufferType::PositionAttrib, MeshShaderCache::Mandatory, "vec3"},
+           {BufferType::ColorAttrib, MeshShaderCache::Optional, "vec4"},
+           {BufferType::RadiiAttrib, MeshShaderCache::Optional, "float"},
+           {BufferType::ScalarMetaAttrib, MeshShaderCache::Optional, "float"}},
+
+          [&](Shader& shader) -> void {
+              shader.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
+              configureShader(shader);
+          })} {
+
+    addPort(inport_);
+    addPort(outport_);
+
+    clipping_.addProperties(clipMode_, clipShadingFactor_, shadeClippedArea_);
+
+    sphereProperties_.addProperties(forceRadius_, defaultRadius_, forceColor_, defaultColor_,
+                                    useMetaColor_, metaColor_);
+
+    addProperties(renderMode_, forceOpaque_, useUniformAlpha_, uniformAlpha_, sphereProperties_,
```

Comment:
Should properties be hidden if corresponding forceX is false? Do they still have any effect if force is false?

Reply:
I really dislike properties appearing and disappearing. Occasionally, I also prefer to be able to modify those in advance before (radius, default color, ...) before enforcing them.

Reply:
I really dislike changing properties that do not have any effect :)
I think that ease of use is most important and hiding irrelevant information helps to reach that goal.

Reply:
Hidden properties have the quirk of not being immediately shown in the link editor though.
Alternative: mark them as read only. As far as I know we have all three approaches in place, which should be unified at some point ;-)

Reply:
Hidden in the Link editor sounds like a bug with an easy fix

Reply:
invisible == hidden (and not updated while the link editor is open #1136)

Reply:
I also think hiding and showing properties is a bad practice. It makes it quite confusing when a property at one point is there and the next it is gone. If you want to hide some properties out then in a composite so there is a consistent and logical way of finding them all the time. One might maybe make them read only when not used. But Martin's usecase of configuring before enabling is also very useful. I also think the boolcomposite can be used an many cases like this